ts/nni_manager/training_service/reusable/environments/localEnvironmentService.ts

```diff
@@ -0,0 +1,140 @@
+import { getExperimentId } from '../../../common/experimentStartupInfo';
+import { getLogger, Logger } from '../../../common/log';
+import { TrialConfigMetadataKey } from '../../common/trialConfigMetadataKey';
+import { PAIClusterConfig } from '../../pai/paiConfig';
+import { NNIPAIK8STrialConfig } from '../../pai/paiK8S/paiK8SConfig';
+import { EnvironmentInformation, EnvironmentService } from '../environment';
+import { StorageService } from '../storageService';
+import { TrialConfig } from '../../common/trialConfig';
+import { getExperimentRootDir, isAlive } from '../../../common/utils';
+import { LocalConfig } from '../../local/localTrainingService';
+import { execMkdir, validateCodeDir, runScript, fileExist, execCopydir } from '../../common/util';
+import { FileCommandChannel } from '../channels/fileCommandChannel';
+import { CommandChannel } from "../commandChannel";
+
+
+@component.Singleton
+export class LocalEnvironmentService extends EnvironmentService {
+
+    private readonly log: Logger = getLogger();
+    private localTrialConfig: TrialConfig | undefined;
+    private localConfig: LocalConfig | undefined;
+    private experimentRootDir: string;
+    private experimentId: string;
+
+    constructor() {
+        super();
+        this.experimentId = getExperimentId();
+        this.experimentRootDir = getExperimentRootDir();
+    }
+
+    public get environmentMaintenceLoopInterval(): number {
+        return 5000;
+    }
+
+    public get hasStorageService(): boolean {
+        return false;
+    }
+
+    public async config(key: string, value: string): Promise<void> {
+        switch (key) {
+            case TrialConfigMetadataKey.LOCAL_CONFIG:
+                this.localConfig = <LocalConfig>JSON.parse(value);
+                break;
+            case TrialConfigMetadataKey.TRIAL_CONFIG:
+                this.localTrialConfig = <TrialConfig>JSON.parse(value);
+                break;
+            default:
+                this.log.debug(`OpenPAI not proccessed metadata key: '${key}', value: '${value}'`);
+        }
+    }
+
+    public async refreshEnvironmentsStatus(environments: EnvironmentInformation[]): Promise<void> {
+        const tasks: Promise<void>[] = [];
+        environments.forEach(async (environment) => {
+            tasks.push(this.refreshEnvironment(environment));
+        });
+        await Promise.all(tasks);
+    }
+
+    public async refreshEnvironment(environment: EnvironmentInformation): Promise<void> {
+        const jobpidPath: string = `${environment.runnerWorkingFolder}/pid`;
+        const runnerReturnCodeFilePath: string = `${environment.runnerWorkingFolder}/code`;
+        /* eslint-disable require-atomic-updates */
+        try {
+            // check if pid file exist
+            const pidExist = await fileExist(jobpidPath);
+            if (!pidExist) {
+                return;
+            }
+            const pid: string = await fs.promises.readFile(jobpidPath, 'utf8');
```

Comment:
According to nodejs' doc, we should read file and handle error instead of check exist before reading.